File: lancersBackEnd/model/Project.py
class Project:
    def __init__(self, idProject: int, ownerEmail: str, name: str, description: str, link: list[str], images: list[str], iframes: list[str]):
        self._idProject = idProject
        self._ownerEmail = ownerEmail
        self._name = name
        self._description = description
        self._link = link
        self._images = images
        self._iframes = iframes

    @property
    def idProject(self) -> int:
        return self._idProject

    @idProject.setter
    def idProject(self, value: int):
        self._idProject = value

    @property
    def ownerEmail(self) -> str:
        return self._ownerEmail

    @ownerEmail.setter
    def ownerEmail(self, value: str):
        self._ownerEmail = value

    @property
    def name(self) -> str:
        return self._name

    @name.setter
    def name(self, value: str):
        self._name = value

    @property
    def description(self) -> str:
        return self._description

    @description.setter
    def description(self, value: str):
        self._description = value

    @property
    def link(self) -> list[str]:
        return self._link

    @link.setter
    def link(self, value: list[str]):
        self._link = value

    @property
    def images(self) -> list[str]:
        return self._images

    @images.setter
    def images(self, value: list[str]):
        self._images = value

    @property
    def iframes(self) -> list[str]:
        return self._iframes

    @iframes.setter
    def iframes(self, value: list[str]):
        self._iframes = value

File: src/model/Project.ts
import {images, link} from "ionicons/icons";

export class Project{
  private _idProject !: number;
  private _ownerEmail !: string;
  private _name !: string;
  private _description !: string;
  private _link !: string[];
  private _images !: string[];
  private _iframes !: string[];


  constructor(idProject: number, ownerEmail: string, name: string, description: string, link: string[], images: string[], iframes: string[]) {
    this._idProject = idProject;
    this._ownerEmail = ownerEmail;
    this._name = name;
    this._description = description;
    this._link = link;
    this._images = images;
    this._iframes = iframes;
  }


  get idProject(): number {
    return this._idProject;
  }

  set idProject(value: number) {
    this._idProject = value;
  }

  get ownerEmail(): string {
    return this._ownerEmail;
  }

  set ownerEmail(value: string) {
    this._ownerEmail = value;
  }

  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }

  get description(): string {
    return this._description;
  }

  set description(value: string) {
    this._description = value;
  }

  get link(): string[] {
    return this._link;
  }

  set link(value: string[]) {
    this._link = value;
  }

  get images(): string[] {
    return this._images;
  }

  set images(value: string[]) {
    this._images = value;
  }

  get iframes(): string[] {
    return this._iframes;
  }

  set iframes(value: string[]) {
    this._iframes = value;
  }
}
